File: bindings/python/ailoy/models/google.py
from typing import Literal

from pydantic import BaseModel, Field

GeminiModelId = (
    Literal[
        "gemini-2.5-flash",
        "gemini-2.5-pro",
        "gemini-2.0-flash",
        "gemini-1.5-flash",
        "gemini-1.5-pro",
    ]
    | str
)


class GeminiModel(BaseModel):
    component_type: str = Field("gemini", frozen=True, init=False)
    id: GeminiModelId
    api_key: str

    def to_attrs(self):
        return {
            "model": self.id,
            "api_key": self.api_key,
        }

File: bindings/js-node/src/models/google.ts
const knownGeminiModelIds = [
  "gemini-2.5-flash",
  "gemini-2.5-pro",
  "gemini-2.0-flash",
  "gemini-1.5-flash",
  "gemini-1.5-pro",
] as const;

export type GeminiModelId =
  | (typeof knownGeminiModelIds)[number]
  | (string & {});

interface GeminiModelArgs {
  id: GeminiModelId;
  apiKey: string;
}

export class GeminiModel {
  id: GeminiModelId;
  apiKey: string;
  readonly componentType: string = "gemini";

  constructor(args: GeminiModelArgs) {
    this.id = args.id;
    this.apiKey = args.apiKey;
  }

  toAttrs() {
    return {
      model: this.id,
      api_key: this.apiKey,
    };
  }
}

export default function (args: GeminiModelArgs): GeminiModel {
  return new GeminiModel(args);
}
